### ISSUE TITLE
Supplier fails when using KPL enabled

### ISSUE THREAD

**Issue by kothapet** [spring-cloud/spring-cloud-stream-binder-aws-kinesis#208]
**In what version(s) of Spring Cloud Stream Binder for AWS Kinesis are you seeing this issue?**

4.0.0

**Describe the bug**

I have simple Supplier method that writes to kinesis, and it works fine with kclKplEnabled=false. Once I make kclKplEnabled=true it fails. Not sure what is causing this.

I am using default credentials ~/.aws/credentials (so guessing its different from https://github.com/spring-cloud/spring-cloud-stream-binder-aws-kinesis/issues/203) 

**To Reproduce**

code

```
@SpringBootApplication
public class ScsKinesisProj1Application {

	public static void main(String[] args) {
		SpringApplication.run(ScsKinesisProj1Application.class, args);
	}

	Supplier<Integer> mySupplier() {
		return() -> {
			Integer int1 = (int) (Math.random() * 1000);
			System.out.println("mySupplier creating payload " + int1);
			return int1;
		};
	}
}
```

properties

```
#
spring.cloud.stream.kinesis.binder.kplKclEnabled=true
#
spring.cloud.function.definition=mySupplier
#
spring.cloud.stream.bindings.mySupplier-out-0.binder=kinesis
spring.cloud.stream.bindings.mySupplier-out-0.destination=my-test-stream
#
#
cloud.aws.stack.auto=false
spring.cloud.aws.region.static=us-east-1
#
```
** error log **

```
2023-09-07T19:44:07.295-04:00  WARN 31770 --- [kpl-daemon-0003] c.a.s.k.producer.LogInputStreamReader    : [2023-09-07 19:44:07.294556] [0x00007c54][0x00007fe21cbfb6c0] [warning] [AWS Log: ERROR](AWSXmlClient)HTTP response code: 403
Resolved remote host IP address: 52.119.198.128
Request ID: 6cc49f27-7911-413c-8ae4-a74c100d6044
Exception name: InvalidClientTokenId
Error message: The security token included in the request is invalid.
4 response headers:
content-length : 306
content-type : text/xml
date : Thu, 07 Sep 2023 23:44:07 GMT
x-amzn-requestid : 6cc49f27-7911-413c-8ae4-a74c100d6044
2023-09-07T19:44:07.296-04:00  INFO 31770 --- [kpl-daemon-0003] c.a.s.k.producer.LogInputStreamReader    : [2023-09-07 19:44:07.294694] [0x00007c54][0x00007fe21cbfb6c0] [info] [AWS Log: WARN](AWSClient)If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.
2023-09-07T19:44:07.297-04:00 ERROR 31770 --- [kpl-daemon-0003] c.a.s.k.producer.LogInputStreamReader    : [2023-09-07 19:44:07.294804] [0x00007c54][0x00007fe21cbfb6c0] [error] [pipeline.h:228] Failed to get StreamARN using STS GetCallerIdentity | Code: InvalidClientTokenId | Message: The security token included in the request is invalid. | Request was: Action=GetCallerIdentity&Version=2011-06-15
2023-09-07T19:44:07.297-04:00  WARN 31770 --- [kpl-daemon-0003] c.a.s.k.producer.LogInputStreamReader    : terminate called after throwing an instance of 'boost::wrapexcept<boost::exception_detail::error_info_injector<boost::log::v2s_mt_posix::system_error> >'
2023-09-07T19:44:07.297-04:00  WARN 31770 --- [kpl-daemon-0003] c.a.s.k.producer.LogInputStreamReader    :   what():  Failed to set TLS value: Invalid argument
mySupplier creating payload 114
2023-09-07T19:44:08.158-04:00 ERROR 31770 --- [   scheduling-1] o.s.integration.handler.LoggingHandler   : org.springframework.messaging.MessageHandlingException: error occurred in message handler [org.springframework.integration.aws.outbound.KplMessageHandler@1af0b111], failedMessage=GenericMessage [payload=byte[3], headers={contentType=application/json, id=e8c9201d-1819-565d-810e-63a6fa8be62a, timestamp=1694130248154}]
	at org.springframework.integration.support.utils.IntegrationUtils.wrapInHandlingExceptionIfNecessary(IntegrationUtils.java:191)
	at org.springframework.integration.handler.AbstractMessageHandler.doHandleMessage(AbstractMessageHandler.java:108)
	at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:73)
	at org.springframework.cloud.stream.binder.AbstractMessageChannelBinder$SendingHandler.handleMessageInternal(AbstractMessageChannelBinder.java:1185)
	at org.springframework.integration.handler.AbstractMessageHandler.doHandleMessage(AbstractMessageHandler.java:105)
	at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:73)
	at org.springframework.integration.dispatcher.AbstractDispatcher.tryOptimizedDispatch(AbstractDispatcher.java:115)
	at org.springframework.integration.dispatcher.UnicastingDispatcher.doDispatch(UnicastingDispatcher.java:133)
	at org.springframework.integration.dispatcher.UnicastingDispatcher.dispatch(UnicastingDispatcher.java:106)
	at org.springframework.integration.channel.AbstractSubscribableChannel.doSend(AbstractSubscribableChannel.java:72)
	at org.springframework.integration.channel.AbstractMessageChannel.sendInternal(AbstractMessageChannel.java:373)
	at org.springframework.integration.channel.AbstractMessageChannel.sendWithMetrics(AbstractMessageChannel.java:344)
	at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:324)
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:187)
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:166)
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:47)
	at org.springframework.messaging.core.AbstractMessageSendingTemplate.send(AbstractMessageSendingTemplate.java:109)
	at org.springframework.integration.router.AbstractMessageRouter.doSend(AbstractMessageRouter.java:217)
	at org.springframework.integration.router.AbstractMessageRouter.handleMessageInternal(AbstractMessageRouter.java:199)
	at org.springframework.integration.handler.AbstractMessageHandler.doHandleMessage(AbstractMessageHandler.java:105)
	at org.springframework.integration.handler.AbstractMessageHandler.handleWithMetrics(AbstractMessageHandler.java:90)
	at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:70)
	at org.springframework.integration.dispatcher.AbstractDispatcher.tryOptimizedDispatch(AbstractDispatcher.java:115)
	at org.springframework.integration.dispatcher.UnicastingDispatcher.doDispatch(UnicastingDispatcher.java:133)
	at org.springframework.integration.dispatcher.UnicastingDispatcher.dispatch(UnicastingDispatcher.java:106)
	at org.springframework.integration.channel.AbstractSubscribableChannel.doSend(AbstractSubscribableChannel.java:72)
	at org.springframework.integration.channel.AbstractMessageChannel.sendInternal(AbstractMessageChannel.java:373)
	at org.springframework.integration.channel.AbstractMessageChannel.sendWithMetrics(AbstractMessageChannel.java:344)
	at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:324)
	at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:297)
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:187)
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:166)
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:47)
	at org.springframework.messaging.core.AbstractMessageSendingTemplate.send(AbstractMessageSendingTemplate.java:109)
	at org.springframework.integration.endpoint.SourcePollingChannelAdapter.handleMessage(SourcePollingChannelAdapter.java:196)
	at org.springframework.integration.endpoint.AbstractPollingEndpoint.messageReceived(AbstractPollingEndpoint.java:474)
	at org.springframework.integration.endpoint.AbstractPollingEndpoint.doPoll(AbstractPollingEndpoint.java:460)
	at org.springframework.integration.endpoint.AbstractPollingEndpoint.pollForMessage(AbstractPollingEndpoint.java:412)
	at org.springframework.integration.endpoint.AbstractPollingEndpoint.lambda$createPoller$4(AbstractPollingEndpoint.java:348)
	at org.springframework.integration.util.ErrorHandlingTaskExecutor.lambda$execute$0(ErrorHandlingTaskExecutor.java:57)
	at org.springframework.core.task.SyncTaskExecutor.execute(SyncTaskExecutor.java:50)
	at org.springframework.integration.util.ErrorHandlingTaskExecutor.execute(ErrorHandlingTaskExecutor.java:55)
	at org.springframework.integration.endpoint.AbstractPollingEndpoint.lambda$createPoller$5(AbstractPollingEndpoint.java:341)
	at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54)
	at org.springframework.scheduling.concurrent.ReschedulingRunnable.run(ReschedulingRunnable.java:96)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.amazonaws.services.kinesis.producer.DaemonException: The child process has been shutdown and can no longer accept messages.
	at com.amazonaws.services.kinesis.producer.Daemon.add(Daemon.java:173)
	at com.amazonaws.services.kinesis.producer.KinesisProducer.flush(KinesisProducer.java:916)
	at com.amazonaws.services.kinesis.producer.KinesisProducer.flush(KinesisProducer.java:936)
	at org.springframework.integration.aws.outbound.KplMessageHandler.handleMessageToAws(KplMessageHandler.java:301)
	at org.springframework.integration.aws.outbound.AbstractAwsMessageHandler.handleMessageInternal(AbstractAwsMessageHandler.java:116)
	at org.springframework.integration.handler.AbstractMessageHandler.doHandleMessage(AbstractMessageHandler.java:105)
	... 49 more

2023-09-07T19:44:08.451-04:00 ERROR 31770 --- [kpl-daemon-0005] c.a.s.kinesis.producer.KinesisProducer   : Error in child process

java.lang.RuntimeException: EOF reached during read
	at com.amazonaws.services.kinesis.producer.Daemon.fatalError(Daemon.java:532) ~[amazon-kinesis-producer-0.15.7.jar:na]
	at com.amazonaws.services.kinesis.producer.Daemon.fatalError(Daemon.java:508) ~[amazon-kinesis-producer-0.15.7.jar:na]
	at com.amazonaws.services.kinesis.producer.Daemon.fatalError(Daemon.java:504) ~[amazon-kinesis-producer-0.15.7.jar:na]
	at com.amazonaws.services.kinesis.producer.Daemon.readSome(Daemon.java:553) ~[amazon-kinesis-producer-0.15.7.jar:na]
	at com.amazonaws.services.kinesis.producer.Daemon.receiveMessage(Daemon.java:243) ~[amazon-kinesis-producer-0.15.7.jar:na]
	at com.amazonaws.services.kinesis.producer.Daemon.access$500(Daemon.java:61) ~[amazon-kinesis-producer-0.15.7.jar:na]
	at com.amazonaws.services.kinesis.producer.Daemon$3.run(Daemon.java:298) ~[amazon-kinesis-producer-0.15.7.jar:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]

```

**Expected behavior**

Should write to kinesis stream.

**Sample**

see above

**Comment by artembilan** [comment]
I think your problem is exactly the same as it is mentioned in that linked issue.
See related SO thread: https://stackoverflow.com/questions/76838474/stswebidentitytokenfilecredentialsprovider-autoconfiguration-not-working-with-sp.

It shows exactly same errors for native PKL library.

I think we are still missing STS credentials mentioned in that SO thread and linked issue.

We will investigate what can be done.

Thanks

**Comment by kothapet** [comment]
Thanks @artembilan

**Comment by artembilan** [comment]
Closed as `Duplicate` of #203
